pkg/controller/sparkapplication/controller.go

```diff
@@ -691,8 +692,16 @@ func (c *Controller) submitSparkApplication(app *v1beta1.SparkApplication) *v1be
 	return app
 }
 
-func (c *Controller) shouldDoBatchScheduling(app *v1beta1.SparkApplication) bool {
-	return c.batchScheduler != nil && c.batchScheduler.ShouldSchedule(app)
+func (c *Controller) shouldDoBatchScheduling(app *v1beta1.SparkApplication) (bool, schedulerinterface.BatchScheduler) {
+	if c.batchSchedulerMgr == nil || app.Spec.BatchScheduler == nil || *app.Spec.BatchScheduler == "" {
```

Comment:
What if `app.Spec.BatchScheduler` is empty, but `app.Spec.Driver.SchedulerName` or `app.Spec.Executor.SchedulerName` is set?

Reply:
Since we have `BatchScheduler` supported in SparkApplication's Spec. I personally would prefer to have only one entrance for end user to trigger batch scheduling. and for the configuration of driver and executor scheduler, they still can utilize them to make SparkApplication scheduled via other scheduler.

Reply:
Makes sense to me.